src/NuGet.Clients/NuGet.Console/PowerConsoleToolWindow.cs

```diff
@@ -51,7 +52,15 @@ private PowerConsoleWindow PowerConsoleWindow
 
         private IVsUIShell VsUIShell
         {
-            get { return this.GetService<IVsUIShell>(typeof(SVsUIShell)); }
+            get
+            {
+                return NuGetUIThreadHelper.JoinableTaskFactory.Run(GetIVsUIShellAsync);
```

Comment:
I think there's a better way to do this. 

The `GetService` call that used to be here is a consequence of this getting invoked on a synchronous codepath.

Changing to JTF.Run remove the GetService call, but doesn't really do anything in the grandscheme of improving perf. 

To improve perf, you should go another layer up. 

For example, this is *only* in https://github.com/NuGet/NuGet.Client/blob/a1fa3375d91d95e617510480f4c3b28324563f94/src/NuGet.Clients/NuGet.Console/PowerConsoleToolWindow.cs#L521. 

Which is already an async codepath, so the *correct* thing to do here would be to call GetServiceAsync *directly*. 

Same applies to *all* situations.